nautobot/dcim/tests/test_models.py

```diff
@@ -197,6 +199,123 @@ def test_interface_template_sets_interface_status(self):
+                secrets_group=self.secrets_groups[0],
+                protocol_group_id="1",
+            ),
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 2",
+                protocol="carp",
+                status=statuses[1],
+                virtual_ip=self.ips[1],
+                secrets_group=self.secrets_groups[1],
+                protocol_group_id="2",
+            ),
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 3",
+                protocol="vrrp",
+                status=statuses[2],
+                virtual_ip=self.ips[2],
+                secrets_group=None,
+                protocol_group_id="3",
+            ),
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 4",
+                protocol="glbp",
+                status=statuses[3],
+                virtual_ip=self.ips[3],
+                secrets_group=self.secrets_groups[2],
+            ),
+        )
+
+        for group in self.interface_redundancy_groups:
+            group.validated_save()
+
+        self.device_type = DeviceType.objects.first()
+        self.device_role = DeviceRole.objects.first()
+        self.site = Site.objects.first()
+        self.device = Device.objects.create(
+            device_type=self.device_type, device_role=self.device_role, name="Device 1", site=self.site
+        )
+        non_default_status = Status.objects.get_for_model(Interface).exclude(name="Active").first()
+        self.interfaces = (
+            Interface.objects.create(
+                device=self.device,
+                name="Interface 1",
+                type="1000base-t",
+                status=non_default_status,
+            ),
+            Interface.objects.create(
+                device=self.device,
+                name="Interface 2",
+                type="1000base-t",
+                status=non_default_status,
+            ),
+            Interface.objects.create(
+                device=self.device,
+                name="Interface 3",
+                type=InterfaceTypeChoices.TYPE_BRIDGE,
+                status=non_default_status,
+            ),
+            Interface.objects.create(
+                device=self.device,
+                name="Interface 4",
+                type=InterfaceTypeChoices.TYPE_1GE_GBIC,
+                status=non_default_status,
+            ),
+            Interface.objects.create(
+                device=self.device,
+                name="Interface 5",
+                type=InterfaceTypeChoices.TYPE_LAG,
+                status=non_default_status,
+            ),
+        )
+
+    def test_add_interface(self):
+        interfaces = Interface.objects.all()
+        interface_redundancy_group = self.interface_redundancy_groups[0]
+        previous_count = InterfaceRedundancyGroupAssociation.objects.filter(
+            interface_redundancy_group=interface_redundancy_group
+        ).count()
```

Comment:
Perhaps also/instead, check `interface_redundancy_group.interfaces.count()` here and below?